### PR TITLE
Move timer gate to common package

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Move timer gate to common package

**Why?**
Code cleanup

**How did you test it?**
N/A

**Potential risks**
No

**Is hotfix candidate?**
No

### CODE REVIEW

File: common/timer/local_gate.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+		// the actual timer which will fire
+		timer *time.Timer
+		// variable indicating when the above timer will fire
+		nextWakeupTime time.Time
+	}
+)
+
+// NewLocalGate create a new timer gate instance
+func NewLocalGate(timeSource clock.TimeSource) LocalGate {
```
Comment (yiminc) [inline common/timer/local_gate.go:53]: github is not able to figure out this is old code moved around? I hate that we lost the history of the code.

**Review by yiminc — APPROVED** [review]

File: common/timer/gate.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+type (
+	// Gate interface
+	Gate interface {
+		// FireChan return the channel which will be fired when time is up
+		FireChan() <-chan struct{}
+		// FireAfter check will the timer get fired after a certain time
+		FireAfter(now time.Time) bool
+		// Update update the timer gate, return true if update is a success
+		// success means timer is idle or timer is set with a sooner time to fire
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply common/timer/gate.go:39]: idle meaning this timer gate will not fire at all

File: common/timer/local_gate.go (hunk 1 above)
Comment (wxing1292) [reply common/timer/local_gate.go:53]: not sure ...

Comment (wxing1292) [reply common/timer/local_gate.go:53]: maybe due to TimerGate => timer.Gate

**Pull request merged**